ntoskrnl/ke/i386/trap.s

```diff
@@ -17,14 +17,28 @@ MACRO(GENERATE_IDT_STUB, Vector)
 idt _KiUnexpectedInterrupt&Vector, INT_32_DPL0
 ENDM
 
+/*
+ * The INT stub handlers must all have the same size (stored in _KiUnexpectedEntrySize)
+ * in all situations. Therefore, we force using a push instruction with 32-bit operand
+ * and a non-short near jump.
+ */
 MACRO(GENERATE_INT_HANDLER, Vector)
-//.func KiUnexpectedInterrupt&Vector
 _KiUnexpectedInterrupt&Vector:
-    /* This is a push instruction with 8bit operand. Since the instruction
-       sign extends the value to 32 bits, we need to offset it */
-    push (Vector - 128)
-    jmp _KiEndUnexpectedRange@0
-//.endfunc
+#ifdef _USE_ML
+    push dword ptr (&Vector)
```

Comment:
Are you sure this generates the correct instruction, and not something like ```push dword ptr ds:[Vector]```?

Reply:
Last time I worked on that, using ML from MSVC 2010, it generated the correct expression, the same as the one hardcoded below for GCC.